test/templates/pagination/component.njk

```diff
@@ -1,50 +1,47 @@
-            {%- if (item === currentPage) %}
-    <a href="#" class="fd-pagination__link">{{item - 1}}</a>
-            {%- elseif (currentPage !== (pages)) or (item !== (pages))%}
-    <a href="#" class="fd-pagination__link">{{currentPage -1}}</a>
-    <a href="#" class="fd-pagination__link" aria-selected="true">{{currentPage}}</a>
-    <a href="#" class="fd-pagination__link">{{currentPage+1}}</a>
-    <span class="fd-pagination__link fd-pagination__link--more" role="presentation"></span>
-            {%- endif %}
-    <a href="#" class="fd-pagination__link"{{ ' aria-selected="true"' if item === currentPage  or item === pages + 1  }}>{{item}}</a>
-        {%- endfor %}
-    {%- else %}
-        {%- for item in range(1, 4) %}
-    <a href="#" class="fd-pagination__link"{{ ' aria-selected="true"' if item === currentPage }}>{{item}}</a>
-        {%- endfor %}
-        {%- for item in range(pages, pages+1) %}
-    <span class="fd-pagination__link fd-pagination__link--more" role="presentation"></span>
-            {%- if (item === currentPage) %}
-    <a href="#" class="fd-pagination__link">{{item - 1}}</a>
-            {%- endif %}
-    <a href="#" class="fd-pagination__link"{{ ' aria-selected="true"' if item === currentPage  or item === pages + 1  }}>{{item}}</a>
-        {%- endfor %}
-    {%- endif %}
+{%- endif %}
+{%- if pages < 4 %}
+    {%- for item in range(1, pages + 1) %}{{pageLink(item, item === currentPage)}}{%- endfor %}
+{%- elseif (currentPage === pages) or (currentPage === pages + 1) or (currentPage > 4) %}
+    {{ pageLink(1) }}
+    {%- for item in range(pages, pages+1) %}
+        <span class="fd-pagination__link fd-pagination__link--more" role="presentation"></span>
+        {%- if (item === currentPage) %}{{pageLink(item - 1)}}
+        {%- elseif (currentPage !== (pages)) or (item !== (pages))%}
+            {{pageLink(currentPage - 1)}}{{pageLink(currentPage, true)}}{{pageLink(currentPage + 1)}}
+            <span class="fd-pagination__link fd-pagination__link--more" role="presentation"></span>
+        {%- endif %} 
+        {{ pageLink(item, item === currentPage  or item === pages + 1) }}
+    {%- endfor %}
+{%- else %}
+    {%- for item in range(1, 4) %}{{pageLink(item, item === currentPage)}}{%- endfor %}
+    {%- for item in range(pages, pages+1) %}
+        <span class="fd-pagination__link fd-pagination__link--more" role="presentation"></span>
+        {%- if (item === currentPage) %} {{pageLink(item -1)}} {%- endif %}
+        {{ pageLink(item, item === currentPage  or item === pages + 1)}}
+    {%- endfor %}
+{%- endif %}
+{%- if currentPage !== pages %}
     <a href="#" class="fd-pagination__link fd-pagination__link--next" aria-label="Next" aria-disabled="{{currentPage === pages | round(0)}}"></a>
+{%- endif %}
   </nav>
 </div>
 {% endmacro %}
+
+{% macro pageLink(page=0, selected=false) %}
+<a href="#" {{' aria-selected="true"' if selected}} class="fd-pagination__link">{{page}}</a>{% endmacro %}
```

Comment:
Can we add `is-selected` class here instead of `aria-selected`? I know `fd-selected` mixin allows for aria-selected, but I want to move away from aria selectors.

Reply:
done